chill-hadoop/src/main/java/com/twitter/chill/hadoop/KryoDeserializer.java

```diff
@@ -46,7 +46,7 @@ public void open(InputStream in) throws IOException {
 
     public Object deserialize(Object o) throws IOException {
         // TODO, we could share these buffers if we see that alloc is bottlenecking
-        byte[] bytes = new byte[inputStream.readInt()];
+        byte[] bytes = new byte[(int) Varint.readUnsignedVarLong(inputStream)];
```

Comment:
I think this flowed out of the kryo api for bytes written returning a long... at the entry point to that i goto an int now, so everything else just ends up being integers